ocs_ci/deployment/acm.py

```diff
@@ -74,15 +76,23 @@ def __init__(self):
     def deploy(self):
         if self.source == "upstream":
             self.deploy_upstream()
-        else:
+        elif self.source == "downstream":
             self.deploy_downstream()
+        else:
+            raise Exception(f"The Submariner source: {self.source} is not recognized")
 
     def deploy_upstream(self):
         self.download_binary()
         self.submariner_configure_upstream()
 
     def deploy_downstream(self):
-        raise NotImplementedError("Deploy downstream functionality not implemented")
+        if self.deployment_type == constants.UI_DEPLOYMENT:
```

Comment:
what if the UI deployment will not be enabled? Then it will not do anything. I would raise some NotImplementedError in the else block.

Reply:
@jimdevops19 , here we need to handle the case, we can have flag which tells what type of submariner deployment is this ? `submariner_ui_deployment` and set it to true. WDYT?

Reply:
Addressed in a way that we pass `ui_deployment` through DEPLOYMENT section, since it is a key consistently passed from this section and we should be consistent with it